tests/test_deprecations.py

```diff
@@ -109,8 +133,14 @@ def test_deprecated_batch_quantity_keyword(campaign):
 
 @pytest.mark.parametrize("flag", (True, False))
 def test_deprecated_strategy_allow_flags(flag):
-    """Using the deprecated strategy "allow" flags raises an error."""
+    """Using the deprecated recommender "allow" flags raises an error."""
     with pytest.raises(DeprecationError):
-        TwoPhaseStrategy(allow_recommending_already_measured=flag)
+        TwoPhaseMetaRecommender(allow_recommending_already_measured=flag)
+    with pytest.raises(DeprecationError):
+        TwoPhaseMetaRecommender(allow_recommending_already_measured=flag)
+
+
+def test_deprecated_strategy_campaign_flag(recommender):
+    """Using the deprecated strategy keyword raises an error."""
     with pytest.raises(DeprecationError):
-        TwoPhaseStrategy(allow_recommending_already_measured=flag)
+        Campaign(None, None, None, strategy=recommender)  # noqa: E999
```

Comment:
What is that `noqa` error here and why is it necessary?

Reply:
hmm i gotta be honest no idea, I thought its evtl because of the None usage here but Ive removed all E999 exceptions and it still works fine for me locally

in the process I discovered another bug in the adjacent test above testing the same flag twice